pf/internal/pfutils/attr.go

```diff
@@ -112,14 +112,22 @@ const (
 	NestingModeMap     NestingMode = 4
 )
 
+// Classifies the results of LookupTerraformPath. The interesting cases:
+//
+// 1. IsAttr=true, and Attr is set; this means an Attribute was found
+// 2. IsBlock=true, and Block is set; this means a Block was found
+// 3. IsNestedObject=true; this means the path is a nested object one level down from Attr or Block
+// 4. IsMisc=true; this groups all other cases, such as resolving to a simple atomic type as String
```

Comment:
Is LookupResult{IsAttr: true} is not valid!

If I had Haskell here it would be `data LookupResult = AttrResult Attr | BlockResult Block | NestedObject | Misc`

Reply:
I miss clear sum types. If not for https://github.com/pulumi/pulumi-terraform-bridge/pull/929 solving the problem for us, I would recommend we model this as 

```go
type LookupResult interface { isLookupResult() }
type LookupAttr struct { attrLike }
type LookupBlock { blockLike }
...
```